crudApp/src/reducers/Crud.js

```diff
@@ -0,0 +1,37 @@
+//A single reducer changeTheState which defines how to perform actions based on the action type.
+
+const initialState = [{
+  id:"",
+  Name:"",
+  Age:"",
+}];
+
+const changeTheState = (state = initialState, action) => {
+  switch (action.type) {
+    case "ADD":
+      return [
+        ...state,
+        {
+          id: action.payloadId,
+          Name: action.payloadName,
+          Age: action.payloadAge
+        }
+      ];
+    case "EDIT": {
+      [...state][action.payloadIndex].Name = action.payloadName;
+      [...state][action.payloadIndex].Age = action.payloadAge;
+      return state;
+    }
+    case "DELETE": {
+      [...state].splice(action.payloadIndex, 1);
+      return state;
+    }
+    case "GET": {
+      return [...state, ...action.payload];
+    }
+    default:
+      return state;
```

Comment:
return the initial state, find the diff between state and initialstate,

Reply:
The initial state defines the state initially provided to make sure the state gets the value of initial state when nothing is provided and the state defines the current state. 
In my example, I want to pass an empty array as an initial state which means there is no data initially to render.
We can specify the type of objects the array should contain by doing : 
const initialState: {id: String, Name: String, Age: String}[] = [];
But since I am not using typescript in this app, I am not sure how to do it.